src/main/java/org/wildfly/security/auth/client/AuthenticationContextConfigurationClient.java

```diff
@@ -134,7 +134,7 @@ public AuthenticationConfiguration getAuthenticationConfiguration(URI uri, Authe
             final IdentityCredentials privateCredentials = securityIdentity.getPrivateCredentials();
             final IdentityCredentials publicCredentials = securityIdentity.getPublicCredentials();
             // private overrides public
-            configuration = configuration.useForwardedIdentity(null).usePrincipal(securityIdentity.getPrincipal()).useCredentials(publicCredentials.with(privateCredentials));
+            configuration = configuration.useForwardedIdentity(securityIdentity.getPrincipal() == AnonymousPrincipal.getInstance() ? forwardSecurityDomain : null).usePrincipal(securityIdentity.getPrincipal()).useCredentials(publicCredentials.with(privateCredentials));
```

Comment:
The point of capturing here is so that the identity sticks with the configuration even if the configuration moves between threads.  It has to be stable after the point of capture.  The code in RemoteOutboundConnectionService then is what is incorrect; I think the original code here is correct.

It is not correct to use the RemoteOutboundConnectionService for propagating identity; it will not work.  The RemoteOutboundConnectionService is purely a legacy construct.  Outbound authentication should be configured in the Elytron subsystem.